packages/service-clients/end-to-end-tests/odsp-client/src/test/OdspClientFactory.ts

```diff
@@ -48,7 +114,7 @@ export function createOdspClient(
 		throw new Error("client id is missing");
 	}
 
-	if (creds.username === undefined || creds.password === undefined) {
+	if (creds.email === undefined || creds.password === undefined) {
 		throw new Error("username or password is missing for login account");
 	}
```

Comment:
IOdspLoginCredentials should never have undefined email or password, according to its typing.  Probably prefer to move this check to `getCredentials()` to ensure the typing is safe throughout.